### PR TITLE
xpla add

### PR DESCRIPTION
**Pull request by park1847**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. The protocol is usually listed within 24 hours of merging the PR
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/xpla: 
        
 
  
    
    
    Error: 
        Unknown chain(s): xpla
        Note: if you think that the chain is correct but missing from our list, please add it to 'projects/helper/chains.json' file

**Comment by llamatester** [comment]
The adapter at projects/xpla exports TVL: 
        
 
 
    xpla-staking              31.17 M
    xpla                      682.75 M
    staking                   31.17 M
    
    total                    682.75 M

File: projects/xpla/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+async function xpla(timestamp, ethBlock, chainBlocks) {
+  const totalSupply = (
+    await axios.get(`https://dimension-lcd.xpla.dev/cosmos/bank/v1beta1/supply/axpla`)
+  ).data.amount.amount;
+
+  const currentCirculationgSupply = (
+    await axios.get(`https://dimension-fcd.xpla.dev/v1/bank/xplasupply/circulating`)
+  ).data;
+
+  const result = totalSupply / 1000000000000000000 - currentCirculationgSupply
```
Comment (g1nt0ki) [inline projects/xpla/index.js:23]: bit confused by what is happening here

File: projects/xpla/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const axios = require('axios');
+
+async function staking(timestamp, ethBlock, chainBlocks) {
```
Comment (g1nt0ki) [inline projects/xpla/index.js:3]: sorry we are not tracking native chain staking for other chains, once I get some time, will fix this and make it as a metric on each chain page

File: projects/xpla/index.js (hunk 1 above)
Comment (park1847) [reply projects/xpla/index.js:23]: photo_2023-05-30_10-57-26

https://explorer.xpla.io/

You can see this item on the Xpla explorer page.
Can't use that code?

File: projects/xpla/index.js (hunk 2 above)
Comment (park1847) [reply projects/xpla/index.js:3]: I'll delete it !

**Comment by llamatester** [comment]
The adapter at projects/xpla exports TVL: 
        
 
 
    xpla                      677.79 M
    
    total                    677.79 M

File: projects/xpla/index.js (hunk 1 above)
Comment (g1nt0ki) [reply projects/xpla/index.js:12]: @park1847 sorry, but you cant use total circulating of your token as tvl, for something to be counted as tvl, it needs to be tokens locked in smart contracts on the chain like a dex or lending protocol.

Comment (park1847) [reply projects/xpla/index.js:12]: B is the result. B can't be the result?

[C = total supply] minus [ A = current circulating supply] is [ B = non-circulating supply].

**Pull request closed without merging**